packages/cozy-viewer/src/Panel/AI/AIAssistantPanel.jsx

```diff
@@ -0,0 +1,206 @@
+import cx from 'classnames'
+import PropTypes from 'prop-types'
+import React, { useCallback, useState, useEffect } from 'react'
+import { useLocation, useNavigate } from 'react-router-dom'
+
+import { useClient } from 'cozy-client'
+import { extractText, chatCompletion } from 'cozy-client/dist/models/ai'
+import { fetchBlobFileById } from 'cozy-client/dist/models/file'
+import logger from 'cozy-logger'
+import Button from 'cozy-ui/transpiled/react/Buttons'
+import Icon from 'cozy-ui/transpiled/react/Icon'
+import IconButton from 'cozy-ui/transpiled/react/IconButton'
+import AssistantIcon from 'cozy-ui/transpiled/react/Icons/Assistant'
+import CopyIcon from 'cozy-ui/transpiled/react/Icons/Copy'
+import CrossMediumIcon from 'cozy-ui/transpiled/react/Icons/CrossMedium'
+import RefreshIcon from 'cozy-ui/transpiled/react/Icons/Refresh'
+import Paper from 'cozy-ui/transpiled/react/Paper'
+import Stack from 'cozy-ui/transpiled/react/Stack'
+import Typography from 'cozy-ui/transpiled/react/Typography'
+import { useAlert } from 'cozy-ui/transpiled/react/providers/Alert'
+import { useI18n } from 'cozy-ui/transpiled/react/providers/I18n'
+
+import { SUMMARY_SYSTEM_PROMPT, getSummaryUserPrompt } from './prompts'
+import styles from './styles.styl'
+import { useViewer } from '../../providers/ViewerProvider'
+
+const AIAssistantPanel = () => {
+  const { t } = useI18n()
+  const client = useClient()
+  const { file, setIsOpenAiAssistant } = useViewer()
+  const { showAlert } = useAlert()
+
+  const [isLoading, setIsLoading] = useState(true)
+  const [summary, setSummary] = useState('')
+  const [error, setError] = useState(null)
+
+  const location = useLocation()
+  const navigate = useNavigate()
+
+  const handleClose = () => {
+    setIsOpenAiAssistant(false)
+    if (location?.state?.showAIAssistant) {
+      navigate('..')
+    }
+  }
+
+  const summarizeFile = async ({ client, file, stream = false, model }) => {
+    try {
+      const fileBlob = await fetchBlobFileById(client, file?._id)
+
+      const textContent = await extractText(client, fileBlob, {
+        name: file.name,
+        mime: file.mime || file.class
+      })
```

Comment:
We could add a safeguard in case there is no page info, assuming 1 token is ~4 characters.

```js
const roughTokensEstimation = (text) => {
  // Assuming 1 token ~ 4 characters
  return Math.ceil(text.length / 4)
}
...
if (roughTokensEstimation(text) > summaryConfig.maxTokens) { 
  // bye bye
}


```